src/Kernel/ClassOrganization.class.st

```diff
@@ -268,16 +265,18 @@ ClassOrganization >> protocolNames [
 	^ self protocols collect: [ :protocol | protocol name ]
 ]
 
-{ #category : #accessing }
-ClassOrganization >> protocols [
+{ #category : #protocol }
+ClassOrganization >> protocolOf: aSelector [
```

Comment:
With the simple name I'm afraid users (myself included) will mistakenly use a CompiledMethod as argument, because I don't care about the selector that much, but I care about the method.

Reply:
A method of a class is always in a protocol even if it is an instance of Protocol named #as yet unclassified.

The case where it returns nil is if we ask the protocol of a method that is not in the class. Instead we should probably raise an error but that changes the current behavior and I prefer do this change later when the protocol management will be cleaner in the image

Reply:
I will rename it into #protocolOfSelector: tomorrow.

Reply:
Another possibility could be to behave correctly if a method is given.

Reply:
Well, if we have a compiled method, I plan to update #protocol to return an instance of protocol instead of a protocol name in the near future. 

I'm not sure it's worth to have a class knowing how to return the protocol of a method when a method will be able to directly return it soon

Reply:
I renamed into #protocolOfSelector: because I don't think the class should have the responsibility of returning the protocol of a method since it should be the responsibility of the method.

Reply:
That makes a lot of sense